monai/losses/unified_focal_loss.py

```diff
@@ -0,0 +1,238 @@
+    """
+
+    def __init__(
+        self,
+        to_onehot_y: bool = False,
+        delta: float = 0.7,
+        gamma: float = 2,
+        epsilon: float = 1e-7,
+        reduction: Union[LossReduction, str] = LossReduction.MEAN,
+    ):
+        """
+        Args:
+            to_onehot_y : whether to convert `y` into the one-hot format. Defaults to False.
+            delta : weight of the backgrand. Defaults to 0.7.
+            gamma : value of the exponent gamma in the definition of the Focal loss  . Defaults to 0.75.
+            epsilon : it's define a very small number each time. simmily smooth value. Defaults to 1e-7.
+        """
+        super().__init__(reduction=LossReduction(reduction).value)
+        self.to_onehot_y = to_onehot_y
+        self.delta = delta
+        self.gamma = gamma
+        self.epsilon = epsilon
+
+    def forward(self, y_pred: torch.Tensor, y_true: torch.Tensor) -> torch.Tensor:
+        n_pred_ch = y_pred.shape[1]
+
+        if self.to_onehot_y:
+            if n_pred_ch == 1:
+                warnings.warn("single channel prediction, `to_onehot_y=True` ignored.")
+            else:
+                y_true = one_hot(y_true, num_classes=n_pred_ch)
+
+        if y_true.shape != y_pred.shape:
+            raise ValueError(f"ground truth has different shape ({y_true.shape}) from input ({y_pred.shape})")
+
+        y_pred = torch.clamp(y_pred, self.epsilon, 1.0 - self.epsilon)
+        cross_entropy = -y_true * torch.log(y_pred)
+
+        back_ce = torch.pow(1 - y_pred[:, 0], self.gamma) * cross_entropy[:, 0]
+        back_ce = (1 - self.delta) * back_ce
+
+        fore_ce = cross_entropy[:, 1]
+        fore_ce = self.delta * fore_ce
+
+        loss = torch.mean(torch.sum(torch.stack([back_ce, fore_ce], dim=1), dim=1))
+        return loss
+
+
+class AsymmetricUnifiedFocalLoss(_Loss):
+    """
+    AsymmetricUnifiedFocalLoss is a variant of Focal Loss.
+
+    Actually, it's only supported for binary image segmentation now
+
+    Reimplementation of the Asymmetric Unified Focal Tversky Loss described in:
+
+    - "Unified Focal Loss: Generalising Dice and Cross Entropy-based Losses to Handle Class Imbalanced Medical Image Segmentation",
+    Michael Yeung, Computerized Medical Imaging and Graphics
+    """
+
+    def __init__(
+        self,
+        to_onehot_y: bool = False,
+        num_classes: int = 1,
+        weight: float = 0.5,
+        gamma: float = 0.5,
+        delta: float = 0.7,
+        reduction: Union[LossReduction, str] = LossReduction.MEAN,
+    ):
+        """
+        Args:
+            to_onehot_y : whether to convert `y` into the one-hot format. Defaults to False.
+            delta : weight of the backgrand. Defaults to 0.7.
+            gamma : value of the exponent gamma in the definition of the Focal loss  . Defaults to 0.75.
+            epsilon : it's define a very small number each time. simmily smooth value. Defaults to 1e-7.

```

Comment:
what is "simmily smooth value"?

```suggestion
            epsilon : it defines a very small number each time. simmily smooth value. Defaults to 1e-7.

```